src/NUnitFramework/framework/CsvData.cs

```diff
@@ -0,0 +1,136 @@
+﻿// ***********************************************************************
+// Copyright (c) 2009-2015 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework.Interfaces;
+using System.IO;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// Import a Comma Separated Value data source
+    /// </summary>
+    public class CsvData : DataSource
+    {
+
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="source">Filename of CSV file</param>
+        public CsvData(string source)
+        {
+            Name = source;
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="source">CSV file</param>
+        /// <param name="rowNum">number of rows to read</param>
+        public CsvData(string source, int rowNum)
```

Comment:
I think the `rowNum` argument and the `RowNumber` property are confusing. If I did not read the comment, I would have assumed it was which row you want to read, not the number of rows. I think it should be `numRows` and `NumberOfRows`. When designing a public API, it is important to use clear, concise names. @CharliePoole and I often go back and forth over naming endlessly :smile: